feat(Communities): Airdrop flow for collectibles

micieslak opened the pull request:
### What does the PR do

Implements Airdrop flow for collectibles according to the design.

- New components:
  - `SequenceColumnLayout` (replacing removed `CurveSeparatorWithText`), `CommunityNewPermissionView` updated
  - `RecipientTypeSelectionDropdown`
- `HoldingsDropdown` updated to handle collectible models with chain/supply specified, available supply validation
- `CommunityNewAirdropView` extended/refactored:
  - `AirdropTokensSelector` integrated along with `HoldingsDropdown` configured to be used with collectibles model containing supply and network info
  - `AirdropRecipientsSelector` integrated
  - `MembersDropdown` integrated to allow selection recipients among community members
  - max number of recipients calculation for selected collectibles
  - marking as red collectibles with insufficient available supply
   - validation if form is properly filled to perform airdrop
- `CommunitySettingsView.qm` - passing members model and adjusted community tokens model to `CommunityAirdropsSettingsPanel`
- `AirdropRecipientsSelector` - added label indicating current / max number of recipients
- `MembersDropdown` - adjusted to the updated design, Add/Update modes added
- New storybook pages: `CommunityNewAirdropView`, `SequenceColumnLayout`

Closes: #9801 

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it
![Screenshot from 2023-04-28 14-12-08](https://user-images.githubusercontent.com/20650004/235144288-b51a100f-381b-4acb-9b9f-4f8b4e18149d.png)
![Screenshot from 2023-04-28 14-07-38](https://user-images.githubusercontent.com/20650004/235143885-708635ad-2821-420e-b31e-86e4d7ea977f.png)
![Screenshot from 2023-04-28 14-07-30](https://user-images.githubusercontent.com/20650004/235143877-1eedbf91-d289-4a13-b085-cfb8614ab302.png)
![Screenshot from 2023-04-28 14-09-38](https://user-images.githubusercontent.com/20650004/235143926-2be86611-c949-4bd7-b8f3-cf80a8643fc3.png)
![Screenshot from 2023-04-28 14-08-22](https://user-images.githubusercontent.com/20650004/235143940-c4c75076-b61e-4b1d-9d03-79277eb0e803.png)
![Screenshot from 2023-04-28 14-08-53](https://user-images.githubusercontent.com/20650004/235143949-c58c5df0-8060-43cb-b1e3-a626f93b0eb8.png)
![Screenshot from 2023-04-28 14-09-15](https://user-images.githubusercontent.com/20650004/235143964-d337f20d-f5b3-4626-83d7-ddcd0985fea1.png)

alexandraB99 reviewed (approved):
lgtm, thanks @micieslak !

caybro reviewed (approved):
Nice! Looks good to me

noeliaSD commented on ui/app/AppLayouts/Chat/controls/community/InlineNetworksComboBox.qml:
```diff
@@ -109,7 +116,9 @@ StatusComboBox {
 
-        amount: !d.oneItem ? instantiator.amount : ""
+        amount: !d.oneItem
+                ? (instantiator.infiniteAmount ? "∞" : instantiator.amount)
```
Maybe we can add the `infinite` symbol as a readonly property in this component since it is used in different places.

noeliaSD reviewed (changes requested):
Great job!

Just some comments here:

- `Bugs?`

1. In `holdings dropdown`, searcher sometimes is broken. It seems to be disabled. I see it in `CommunityPermissionsSettingsPage` and also in `airdrop` related components, but I cannot reproduce it always. 
2. Maybe here it is needed some model refactor in storybook (chain id property)?
![Screenshot 2023-05-03 at 11 46 29](https://user-images.githubusercontent.com/97019400/235890834-79f1c7e7-9492-4a39-a07e-6f14645da0c4.png)

Here some warnings, I think, due to the model refactor needed:

```
<Unknown File>: networkText is undefined. Adding an object with a undefined member does not create a role for it.
<Unknown File>: supply is undefined. Adding an object with a undefined member does not create a role for it.
<Unknown File>: infiniteSupply is undefined. Adding an object with a undefined member does not create a role for it.
status-desktop/ui/StatusQ/src/StatusQ/Components/StatusRoundedImage.qml:26:5: QML StatusImage: Cannot open: file:///Users/nsantos/status/src/dev/status-desktop/ui/imports/assets/icons/undefined.svg
status-desktop/ui/app/AppLayouts/Chat/controls/community/AirdropTokensSelector.qml:87:21: Unable to assign [undefined] to QString
status-desktop/ui/app/AppLayouts/Chat/views/communities/CommunityNewAirdropView.qml:83:13: Unable to assign [undefined] to bool
status-desktop/ui/app/AppLayouts/Chat/views/communities/CommunityNewAirdropView.qml:81:13: Unable to assign [undefined] to int
```

- `ETH addressess input` issues comparing with design:

1. The top and button margins when the input is empty and just showing the placeholder seem to be higher than in design.
2. Once some addresses have already been added, I'm not sure that the placeholder should be displayed. In designs it seems only the cursor is present.

- `Questions / suggestions:`

1. How can we reach this point if we firstly click on add `Eth address` but we don't add any address? https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=22647-497754&t=jWfhzUpWUpSZU2K6-0 Maybe we need to enable the `esc` key to remove the eth address input component?!
I see in the `AirdropRecipientsSelector`, that by removing all the entries in the addresses input, you can reach the point, but it doesn't have the same behavior in `CommunityNewAirdropView`.

2. In the members dropdown, sometimes it is a bit difficult to select a new member bc the scrollbar is very close to the checkbox. Could it be possible to move the scrollbar more to the right?

3. I'm wondering if this error should be different since the cause is that there are no invalid addresses to airdrop.

![Screenshot 2023-05-03 at 11 41 01](https://user-images.githubusercontent.com/97019400/235890904-37e32376-518c-4712-b002-b3a9f26aa4e9.png)

micieslak commented:
> * `Bugs?`
> 
> 1. In `holdings dropdown`, searcher sometimes is broken. It seems to be disabled. I see it in `CommunityPermissionsSettingsPage` and also in `airdrop` related components, but I cannot reproduce it always.

Trying in various ways but I cannot reproduce. Could you provide some additional details?

> 2. Maybe here it is needed some model refactor in storybook (chain id property)?
>    ![Screenshot 2023-05-03 at 11 46 29](https://user-images.githubusercontent.com/97019400/235890834-79f1c7e7-9492-4a39-a07e-6f14645da0c4.png)

Fixed

> * `ETH addressess input` issues comparing with design:
> 
> 1. The top and button margins when the input is empty and just showing the placeholder seem to be higher than in design.

I fixed the margins of the "+" button. I'm not sure what's wrong with top margin. 

> 2. Once some addresses have already been added, I'm not sure that the placeholder should be displayed. In designs it seems only the cursor is present.

Fixed.

> * `Questions / suggestions:`
> 
> 1. How can we reach this point if we firstly click on add `Eth address` but we don't add any address? https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=22647-497754&t=jWfhzUpWUpSZU2K6-0 Maybe we need to enable the `esc` key to remove the eth address input component?!
>    I see in the `AirdropRecipientsSelector`, that by removing all the entries in the addresses input, you can reach the point, but it doesn't have the same behavior in `CommunityNewAirdropView`.

According to the design, the address input once initialized is always visible. Current implementation also works this way. I hope it's ok.

> 2. In the members dropdown, sometimes it is a bit difficult to select a new member bc the scrollbar is very close to the checkbox. Could it be possible to move the scrollbar more to the right?

Hmm, the whole delegate is clickable so clicking seems to be easy. I'm not sure if sth is wrong here.

> 3. I'm wondering if this error should be different since the cause is that there are no invalid addresses to airdrop.
> 
> ![Screenshot 2023-05-03 at 11 41 01](https://user-images.githubusercontent.com/97019400/235890904-37e32376-518c-4712-b002-b3a9f26aa4e9.png)

'0' as a number of available recipients was improperly calculated in this Storybook page, it's fixed.

noeliaSD commented:
> > 
> > 1. In `holdings dropdown`, searcher sometimes is broken. It seems to be disabled. I see it in `CommunityPermissionsSettingsPage` and also in `airdrop` related components, but I cannot reproduce it always.
> 
> Trying in various ways but I cannot reproduce. Could you provide some additional details?
> 

Here it is a video where I could reproduce it.. (first 2 seconds, I was clicking on the searcher and trying to input something, but no response after clicking neither after keypress) but again, when I tried to reproduce.. I didn't have success so.. no clear steps to follow. When I could reproduce it, it was the first time I run storybook and went to the dropdown. It makes me feel there's something with the focus.. Maybe?¿?

https://user-images.githubusercontent.com/97019400/236255041-863525ca-0941-431f-a130-3014db2ded9e.mov

> 
> > * `ETH addressess input` issues comparing with design:
> > 
> > 
> > 1. The top and bottom margins when the input is empty and just showing the placeholder seem to be higher than in design.
> 
> I fixed the margins of the "+" button. I'm not sure what's wrong with top margin.

I mean this difference (first, design; second, implementation):

![Screenshot 2023-05-04 at 17 21 15](https://user-images.githubusercontent.com/97019400/236256123-6d688cda-eec0-4b20-b901-66736b870565.png)

![Screenshot 2023-05-04 at 17 21 25](https://user-images.githubusercontent.com/97019400/236257405-5145c78b-f534-4e7e-9cfe-cc320fd86090.png)

> > * `Questions / suggestions:`
> > 
> > 
> > 1. How can we reach this point if we firstly click on add `Eth address` but we don't add any address? https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=22647-497754&t=jWfhzUpWUpSZU2K6-0 Maybe we need to enable the `esc` key to remove the eth address input component?!
> >    I see in the `AirdropRecipientsSelector`, that by removing all the entries in the addresses input, you can reach the point, but it doesn't have the same behavior in `CommunityNewAirdropView`.
> 
> According to the design, the address input once initialized is always visible. Current implementation also works this way. I hope it's ok.
> 

Ok!

> > 2. In the members dropdown, sometimes it is a bit difficult to select a new member bc the scrollbar is very close to the checkbox. Could it be possible to move the scrollbar more to the right?
> 
> Hmm, the whole delegate is clickable so clicking seems to be easy. I'm not sure if sth is wrong here.
> 

Here a video for better understanding (when you see the cursor on the checkbox, I'm all the time clicking, and after several clicks, then it is checked):

https://user-images.githubusercontent.com/97019400/236256840-a97267a7-b074-41f2-9d36-1c7005828c0e.mov

caybro commented:
@micieslak one of the community e2e tests is failing, looks like sth legit which needs fixing

micieslak commented:
@noeliaSD, regarding your comments:
1. Locked search - I still cannot reproduce - could you create a bug for that? Seems that it requires further investigation e.g. if only mac is affected.
2. Margins - fixed
3. Members list check boxes - I made the scroll bar thinner, please check if it helps (for me it was working fine anyway). If it doesn't help for you, the source of the problem may be different.

noeliaSD commented:
> @noeliaSD, regarding your comments:
> 
> 1. Locked search - I still cannot reproduce - could you create a bug for that? Seems that it requires further investigation e.g. if only mac is affected.

https://github.com/status-im/status-desktop/issues/10561

> 2. Margins - fixed

Looks good!!

> 3. Members list check boxes - I made the scroll bar thinner, please check if it helps (for me it was working fine anyway). If it doesn't help for you, the source of the problem may be different.

Yeah, now it is comfortable to select an item!! Thanks for the fix!
